### DISCUSSION TITLE
How to prevent entries repetition?

### DISCUSSION THREAD

**Discussion by khomutovspace** [datenstrom/community#455] [Ask a question] [answered]
Hello there! The problem is about getting pages. How to show recommended posts and avoid showing post which is already opened? Or to show some new posts on main page and after that don't show them in category block, also if they are fit it by the filter.

**Comment by schulle4u** [comment]
This is possible, however require some API coding. You can use `$this->yellow->page->isActive()` to check if a pag is within current request. Have a look at the [default navigation layout](https://github.com/datenstrom/yellow/blob/master/system/layouts/navigation.html), the function is used there to provide a CSS class and accessibility information for the currently active page. It should be possible to use this in other layout snippets and simply omit the active page from page collection. 

HTH
Steffen

**Comment by khomutovspace** [comment]
> This is possible, however require some API coding. You can use `$this->yellow->page->isActive()` to check if a pag is within current request. Have a look at the [default navigation layout](https://github.com/datenstrom/yellow/blob/master/system/layouts/navigation.html), the function is used there to provide a CSS class and accessibility information for the currently active page. It should be possible to use this in other layout snippets and simply omit the active page from page collection.
> 
> HTH
> Steffen

Thank you for the answer!

**Comment by khomutovspace** [comment]
There is a way to set a class according to active request, but can you help me to find a way to avoid its' showng?

**Comment by schulle4u** [comment]
I would simply return an empty string if page is currently active. The following foreach statement should work, however I haven't tested it. 

```
<ul>
<?php foreach ($pages as $page): ?>
<?php echo $page->isActive() ? "" : "<li><a href=\"".$page->getLocation(true)."\">".$page->getHtml("titleNavigation")."</a></li>" ?>
<?php endforeach ?>
</ul>
```

**Comment by khomutovspace** [comment]
Can you help me to find away to exclude it from page collection? I mean, not just to give it a CSS class to show it is active, but to solve the problem, when the same posts are showing at the same time at one page in different places. I really need this for my blog.

**Comment by BsNoSi** [comment]
To understand your challenge, I have some questions:
- Why is it necessary to hide messages in several places in a blog to respective visitors depending on their done readings?
- How shall Yellow identify the user to hide respective messages he/she has read on different pages ?

To solve this for a session, the common way is setting cookies and proofing against them.  This mechanismen is (afaik) currently not available. Your welcome to intruduce your plugin.

Another option may be to offer this for logged in readers. In this case depending to your needs a [rights management](https://github.com/datenstrom/yellow/issues/460) would be useful that is extended with a "read" list that servives sessions. I think this would be a better feature but is much more to do for. It is "in discussion".

**Comment by khomutovspace** [comment]
Thank you for the answer! But my need is much easier. I just have the main page of my blog. There are for example 3 latest posts at first and then some categories with 3 posts in each of them too. My task is to hide post from the category, which is suit by the filter also, if if was already shown in "latest" and show if it was not.

**Comment by BsNoSi** [comment]
Why are there more than one list?

Sum up all an show only the latest 5…

**Comment by khomutovspace** [comment]
That's the idea. For example, it can be suitable for the blog. Like, you show the post and after that you show 5 most popular another posts. And using the usual tools, it can show you the post you are already reading, cause you are now right on it's page. My task is to prevent it.

The same with the main page. I want to show some latest pages at first. And then show pages by categories, and some of them can be latest and suit for that category at the same time. But I don't want to show it twice in the same page.

**Comment by markseuffert** [comment]
I checked what's available in the API. There's a method `$pages->merge()` to merge elements of two collections. Perhaps a method is needed that removes elements, which are present in another collection. Is this what you are looking for?

**Comment by khomutovspace** [comment]
Is there any real examples of using this method?

**Comment by markseuffert** [comment, marked as answer]
Now there's support for `diff()` in the [latest core](https://github.com/annaesvensson/yellow-core). Here's an example: 

```
$latest = $this->yellow->content->index()->sort("modified", false)->limit(3);
$popular = $this->yellow->content->index()->filter("tag", "popular")->diff($latest);
```

First line gets a collection of pages, sorts them by modification date, then limits them to three. Second line gets a collection of pages, filters them by tag, then removes latest pages. The name "diff" comes from set theory. The set difference of `{1, 2, 3, 4, 5}` and `{2, 3, 4}` is `{1, 5}`.

**Comment by khomutovspace** [comment]
Does that mean you've just uploaded a new core with that feature? That is brilliant. Right what I was looking for! Thanks a lot!

**Comment by khomutovspace** [comment]
Can I just replace only core file in my already existing project to continue work with that new feature?

**Comment by markseuffert** [comment]
Yes, follow the install instructions or replace your `system/extensions/core.php` file.
